internal/service/vpc/datasource_project_vpc.go

```diff
@@ -12,12 +16,45 @@ import (
 )
 
 func DatasourceProjectVPC() *schema.Resource {
-	aivenProjectVPCDataSourceSchema := schemautil.ResourceSchemaAsDatasourceSchema(aivenProjectVPCSchema,
-		"project", "cloud_name")
-	aivenProjectVPCDataSourceSchema["id"] = &schema.Schema{
-		Type:        schema.TypeString,
-		Description: "ID of the VPC. This can be used to filter out the specific VPC if there are more than one datasource returned.",
-		Optional:    true,
+	aivenProjectVPCDataSourceSchema := map[string]*schema.Schema{
+		"project": {
+			Type:          schema.TypeString,
+			ValidateFunc:  validation.StringMatch(regexp.MustCompile("^[a-zA-Z0-9_-]*$"), "project name should be alphanumeric"),
```

Comment:
nit: I would leave that for backend. Or go client if it's made from schema. This regexp covers basic validation, it even works with empty strings and leading dash/underscore or just dash-only. It doesn't cover length validation either.

Reply:
this is supposed to prevent being rate limited by the backend and to reduce waiting time for the backend to report validation problem, and is actually supposed to be simple enough to cover most cases

backend is going to fail anyway with empty strings, so I believe this could be left as is

Reply:
As @Serpentiel mentioned we already have this validation on the back-end and this is done to communicate the correct formatting on the `terraform plan` stage.